spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/handler/TaskJobLauncherCommandLineRunnerTests.java

```diff
@@ -65,16 +67,13 @@ public void tearDown() {
 
 	@Test
 	public void testTaskJobLauncherCLRSuccessFail() {
-		String[] enabledArgs = new String[] { "--spring.cloud.task.batch.failOnJobFailure=true" };
+		String[] enabledArgs = new String[] {
+				"--spring.cloud.task.batch.failOnJobFailure=true",
+				"--spring.batch.job.enabled=false"};// batch job enabled is turned to false so that we can use Task's JobLauncher
```

Comment:
More importantly, what happened between 2.0 and 2.1 ;-)    In short we have 2 beans vying for the JobLauncher, the one provided by boot and the one we provide if the user wants a Task to fail the execution when  a batch job fails.  Since we don't have bean overriding by default, we have to disable the default boot provided JobLauncher

Reply:
That doesn't answer what changed...we had that feature before, right? This test implies that going forward, users of this functionality will need to set this property (which is a different behavior from the way we did this before).

Reply:
We were using bean overriding before.  If a user set the failOnJobFailure to true we would override the JobLauncher provided by boot.  As of 2.1 we it will not allow that unless we set the flag to re-enable bean overriding or we disable boots default joblauncher.

Reply:
So we're saying that this breaks the existing functionality?

Reply:
yep.  Unless they add one of the two flags.

Reply:
Hmmm....We should talk about this.

Reply:
Agreed

Reply:
agreed